test/setup-ca-kubernetes.sh

```diff
@@ -63,12 +61,16 @@ EOF
     echo "Created: $CSR_NAME.crt"
 }
 
-mkdir -p $WORKDIR 2> /dev/null && cd $WORKDIR
-trap cleaup INT
+# Directory to use for storing intermediate files
+WORKDIR=`mktemp -d -t pmem-XXXX`
+pushd $WORKDIR >/dev/null 2>&1
```

Comment:
Now by adding `pushd` and `popd` this script demands bash shell, would be good to add `!/bin/bash`

Reply:
I missed this, true.
Instead of turning shell to bash, we can consider alternatives:
at least these 2 non-bash methods exist to get back to previous dir without using "cd ..":
1. 
```
cd some-dir
  # do something
cd - > /dev/null
```
2.
```
cd some-dir
  # do something
cd "$OLDPWD"
```

Reply:
but I need to check do all these method work when combined with trap handler

Reply:
In my opinion, both should work.

Reply:
The simplest solution is to do "cd" only in subshells and never attempt to return to the original dir.

Reply:
yes, that is good guidance, but in current script the `cd into` part is in script body and  `cd out` is in cleanup handler function, so it becomes bit more complex. I think the original idea of cleanup handler was to provide removal in case where user stops long-running script with Ctrl-C for example. But when we think about it, script probably was made part of fully autonomous setup after that, so we dont have user any more who stops it, right, so can we simplify?

Reply:
also, that whole idea of "cd out" before removal of directory is bit of purist approach, as the end result seems to be very same if we dont care too much and remove directory where we are in. It is last op in script anyway. What we try to achieve here is extra level of correctness.